### PR TITLE
libnetwork/netutils: minor cleanups and fixes

### PR DESCRIPTION
**Pull request by thaJeztah**
replaces https://github.com/moby/libnetwork/pull/2590
closes https://github.com/moby/libnetwork/pull/2590

### CODE REVIEW

**Review by tianon — APPROVED** [review]

File: netutils/utils_linux_test.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package netutils
```
Comment (thaJeztah) [inline netutils/utils_linux_test.go:1]: oh! this didn't go well; looks like the file was already renamed

File: libnetwork/netutils/utils_linux_test.go [hunk 2]
```diff
@@ -40,29 +40,6 @@ func TestOverlappingNameservers(t *testing.T) {
...
-		t.Fatal(err)
-	}
-
-	_, netX, _ = net.ParseCIDR("10.0.2.0/24")
-	if err := CheckRouteOverlaps(netX); err == nil {
-		t.Fatal("10.0.2.0/24 and 10.0.2.0 should overlap but it doesn't")
-	}
-}
-
 func TestCheckNameserverOverlaps(t *testing.T) {
```
Comment (thaJeztah) [inline libnetwork/netutils/utils_linux_test.go:43]: I _think_ all the other tests should actually be ok on other platforms, so reverting the rename of this file and applying the change I had in https://github.com/moby/libnetwork/pull/2590.

If that explodes, I'll fix it

Comment (thaJeztah) [reply libnetwork/netutils/utils_linux_test.go]: Ah, some other tests also use `netns`. Let me just drop the first commit

**Review by samuelkarp — APPROVED** [review]
LGTM

**Pull request merged**